src/krbd.cc

```diff
@@ -764,3 +764,58 @@ int krbd_showmapped(struct krbd_ctx *ctx, Formatter *f)
 {
   return dump_images(ctx, f);
 }
+
+int krbd_is_image_mapped(struct krbd_ctx *ctx, const char *poolname, 
+                       const char *imgname, const char *snapname,
+                       std::ostringstream &mapped_info) {
+  struct udev_enumerate *enm;
+  struct udev_list_entry *l;
+  struct udev *udev = ctx->udev; 
+  const char *mapped_id, *mapped_pool, *mapped_image, *mapped_snap;
+  int r = 1;
```

Comment:
`*is_protected =
	(info->protection_status == RBD_PROTECTION_STATUS_PROTECTED);
      return 0;`
this function use a boolean pointer save  the protection status, and ret 0 means the function end succesfully.
I intend to use ret value indicate that image mapped or not. I prefer krbd_is_image_NOT_mapped(), though seems a little bit odd...